src/analytics/ledger/address_balance.rs

```diff
@@ -16,19 +19,22 @@ pub struct AddressBalanceAnalytics {
 
 impl AddressBalanceAnalytics {
     /// Initialize the analytics be reading the current ledger state.
-    pub fn init<'a>(unspent_outputs: impl Iterator<Item = &'a LedgerOutput>) -> Self {
-        let mut addresses: HashMap<Address, OutputAmount> = HashMap::new();
-        for output in unspent_outputs {
-            if let Some(&a) = output.owning_address() {
-                *addresses.entry(a).or_default() += output.amount();
-            }
+    pub fn init<'a>(unspent_outputs: impl IntoIterator<Item = &'a LedgerOutput>) -> Self {
+        Self {
+            addresses: unspent_outputs
+                .into_iter()
+                .fold(Default::default(), |mut addresses, output| {
+                    if let Some(&a) = output.owning_address() {
+                        *addresses.entry(a).or_default() += output.amount();
+                    }
+                    addresses
+                }),
```

Comment:
[nitpick] I find that much harder to read.

Reply:
@Alex6323 If you agree I'll change it back.